chain-storage/src/lib.rs

```diff
@@ -294,89 +223,9 @@ impl Storage {
 
     pub fn persist_write(&mut self) -> std::io::Result<()> {
         if let Some(dbtx) = self.current_tx.take() {
-            self.temp_tx_db.clear();
             self.db.write(dbtx)
         } else {
             Ok(())
         }
     }
 }
-
-#[cfg(test)]
-mod test {
```

Comment:
could there be similar tests in buffer? currently, the tests in buffer only use `kvdb_memorydb` which has different semantics "on drop" (or potentially sigterm)

Reply:
> could there be similar tests in buffer? currently, the tests in buffer only use `kvdb_memorydb` which has different semantics "on drop" (or potentially sigterm?)

Do you mean different semantics when dropped without calling `persist_write`?

Reply:
probably not relevant, as the top-level apis changed